docs/contributor/arch/018-switch-from-gateways-to-a-central-agent.md

```diff
@@ -0,0 +1,72 @@
+- Istio is mandatory for mTLS
+  App to Gateway communication is across nodes and with that should use mTLS, so Istio needs to be used which sometimes is not possible. Also, Istio causes additional overhead like noise filtering
+- cumulativetodelta transformation not suppoerted
+  This transformation is mandatory for Dynatrace support and requires that data from a pod gets processed always by the same instance
+- General misconception
+  Gateways are usually running outside the cluster as a first entry point. The cluster should get rid of the data as fast as possible as persistence cannot be provided here (thinks like Kafka would again run outside). Try to compensate for short-lived hickups but otherwise just point back the backpressure to the client and do not try to solve the big picture
+
+## Proposal
+
+Most of the drawbacks can be solved by running the gateway logic node-local only. Every instance always processes only data of the local node. With that a natural scalability is given and can be extended by vertical scaling capabilities. The "old" gateways will be running in agent mode but still called "OTLP Gateway"
+
+![arch](./../assets/otlp-gateway-new.drawio.svg)
+
+The left drawbacks of the agent approach are:
+- coupling of signal types
+  The decoupling of the signal types will be lost, potentially leading to situations where the agent drops logs caused by metrics overload. However, with the right hardening that should not be a problem
+- Downtimes on updates
+  A rollout of the agent cannot happen in a rolling way, especially of the host port assignment. So there will be a downtime always which should be compoensated by the retry mechanism in the otel-sdk
+- host port conflicts
+  The user might serve already the OTLP ports on every port
+
+Additional benefits not outlined yet
+- persistent queue everywhere
+  A small persistent buffer could be used in all components based on the node file system
+- No Istio dependency
+  There is no Istio dependency at all anymore, no at the App and not at the components. However, the metric agent should still support scraping of istiofied endpoints
+
+## Migration considerations
+
+Unifying the three gateways into one agent should be possible without attention by the user. The API stays compatible, and there should be no direct exposure of the architecture to the user. Still, two problems need to be addressed:
+- Push endpoint compatibility
+  It is not possible to simply switch to t localhost push endpoint, the URLs are used in running software via env variables. So they must stay compatible. Luckily, this can be achieved easily by adjusting the services to point to the new agents using the [internalTrafficPolicy: local](https://kubernetes.io/docs/reference/networking/virtual-ips/#internal-traffic-policy) on the service
```

Comment:
```suggestion

  It is not possible to simply switch to the localhost push endpoint. The URLs are used in running software via env variables so they must stay compatible. Luckily, this can be achieved easily by adjusting the services to point to the new agents using the [internalTrafficPolicy: local](https://kubernetes.io/docs/reference/networking/virtual-ips/#internal-traffic-policy) on the service.
```